google-cloud-spanner/lib/google/cloud/spanner/interval.rb

```diff
@@ -0,0 +1,249 @@
+# Copyright 2017 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Google
+  module Cloud
+    module Spanner
+      ##
+      # # Interval
+      #
+      # Represents an interval of time by storing the time components
+      # in months, days and nanoseconds.
+      #
+      # @example
+      #   require "google/cloud/spanner"
+      #
+      #   iso_8601_string = "P1Y2M3DT4H5M6S"
+      #   interval = Google::Cloud::Spanner::Interval::parse iso_8601_string
+      #
+      #   puts interval # "P1Y2M3DT4H5M6S"
+      class Interval
+        NANOSECONDS_IN_A_SECOND = 1000000000
```

Comment:
The style checker will complain here (there should be underscores in the large number for every thousand, i.e. `1_000_000_000`) and in a bunch of other places in the code. You can see the style checker output here: https://github.com/googleapis/ruby-spanner/actions/runs/14457103516/job/40542595879?pr=150. You can also run it locally by moving into the google-cloud-spanner directory and running `toys rubocop`. Please fix all the items flagged.